code/modules/projectiles/updated_projectiles/guns/specialist.dm

```diff
@@ -40,6 +46,200 @@
 		update_attachables()
 		S.icon_state = initial(S.icon_state)
 
+/obj/item/weapon/gun/rifle/sniper/M42A/Fire(atom/target, mob/living/user, params, reflex = 0, dual_wield)
+	if(!able_to_fire(user))
+		return
+	if(targetlaser_on)
+		if(istype(target,/mob) )
+			if(laser_target)
+				laser_target.vis_contents -= laser
+			cdel(laser)
+			to_chat(user, "<span class='danger'>You focus your targeting laser on [target]!</span>")
+			laser_target = target
+			var/obj/effect/overlay/temp/sniper_laser/LT = new (laser_target, "targeting laser")
+			laser = LT
+			targetlaser_on = FALSE
+			laser_target.vis_contents += laser
+			processing_objects.Remove(src) //So we don't accumulate additional processing.
```

Comment:
Wouldn't this refer to the previous laser?

Reply:
No.

The previous laser is handled and removed by the code above.

Reply:
This logic is a little muddy. Is the intention here to switch from one target to another?
Why is this removing one laser if one exists and creating a new one? If it is to refresh it in case the target is different, it should check that and only do it when it is.